src/realm/query.cpp

```diff
@@ -868,9 +868,6 @@ R Query::aggregate(R (ColType::*aggregateMethod)(size_t start, size_t end, size_
                    size_t column_ndx, size_t* resultcount, size_t start, size_t end, size_t limit,
                    size_t* return_ndx) const
 {
-    if (!m_table->is_attached()) {
```

Comment:
The following call to `m_table->is_degenerate()` will throw is the table is detached. This makes sure that every time we test for degeneration, we also test for it being detached.